frontend/src/components/widgets/CameraInput/WebcamComponent.tsx

```diff
@@ -0,0 +1,174 @@
+import { isMobile } from "react-device-detect"
+import Webcam from "react-webcam"
+import { useTheme } from "emotion-theming"
+import { Theme } from "src/theme"
+
+import Button, { Kind } from "src/components/shared/Button"
+import Tooltip, { Placement } from "src/components/shared/Tooltip"
+import Icon from "src/components/shared/Icon"
+import themeColors from "src/theme/baseTheme/themeColors"
+import CameraInputButton from "./CameraInputButton"
+import {
+  StyledBox,
+  StyledCameraInput,
+  StyledDescription,
+  StyledLink,
+  StyledSwitchFacingModeButton,
+} from "./styled-components"
+
+export interface Props {
+  handleCapture: (capturedPhoto: string | null) => void
+  width: number
+  disabled: boolean
+  inClearPhotoMode: boolean
+  setInClearPhotoMode: (inClearPhotoMode: boolean) => void
+}
+
+enum FacingMode {
+  USER = "user",
+  ENVIRONMENT = "environment",
+}
+
+enum WebcamPermission {
+  PENDING = "pending",
+  SUCCESS = "success",
+  ERROR = "error",
+}
+
+const WebcamComponent = ({
+  handleCapture,
+  width,
+  disabled,
+  inClearPhotoMode,
+  setInClearPhotoMode,
+}: Props): ReactElement => {
+  const [webcamPermission, setWebcamRequestState] = useState(
+    WebcamPermission.PENDING
+  )
+  const videoRef = useRef<Webcam>(null)
+  const [facingMode, setFacingMode] = useState(FacingMode.USER)
+
+  function capture(): void {
+    if (videoRef.current !== null) {
+      const imageSrc = videoRef.current.getScreenshot()
+      handleCapture(imageSrc)
+    }
+  }
+
+  function switchCamera(): void {
+    setFacingMode(prevState =>
+      prevState === FacingMode.USER ? FacingMode.ENVIRONMENT : FacingMode.USER
+    )
+  }
+
+  const theme: Theme = useTheme()
+
+  return (
+    <StyledCameraInput width={width}>
+      {webcamPermission !== WebcamPermission.SUCCESS &&
+      !disabled &&
+      !inClearPhotoMode ? (
+        <AskForCameraPermission width={width} />
+      ) : (
+        isMobile && (
+          <StyledSwitchFacingModeButton>
+            <Tooltip content={"Switch camera"} placement={Placement.TOP_RIGHT}>
+              <Button kind={Kind.MINIMAL} onClick={switchCamera}>
+                <Icon
+                  content={SwitchCamera}
+                  size="twoXL"
+                  color={themeColors.white}
+                />
+              </Button>
+            </Tooltip>
+          </StyledSwitchFacingModeButton>
+        )
+      )}
+      <StyledBox
+        hidden={
+          webcamPermission !== WebcamPermission.SUCCESS &&
+          !disabled &&
+          !inClearPhotoMode
+        }
+        width={width}
+      >
+        {!disabled && (
+          <Webcam
+            audio={false}
+            ref={videoRef}
+            screenshotFormat="image/jpeg"
+            screenshotQuality={1}
+            width={width}
+            style={{
+              borderRadius: `${theme.radii.md} ${theme.radii.md} 0 0`,
+            }}
+            height={(width * 9) / 16}
```

Comment:
nit: would recommend computing this and saving it as a `const` above the `return`.

Also, it's not immediately clear to me how the math for this height computation works, so it would be good to add a clarifying comment above the new variable